### PR TITLE
Fix ASV crashes

### PR DESCRIPTION
**Pull request by jiaweig-amzn**
Ran the full test suit without crashing.
There are still some failed tests that will be fixed later.

Signed-off-by: jiaweig <51759646+jiaweig-amzn@users.noreply.github.com>

### CODE REVIEW

File: Gem/Code/Source/MSAA_RPI_ExampleComponent.cpp [hunk 1]
```diff
@@ -101,6 +101,8 @@ namespace AtomSampleViewer
             SampleComponentManagerRequestBus::Broadcast(&SampleComponentManagerRequests::SetNumMSAASamples, m_numSamples);
             SampleComponentManagerRequestBus::Broadcast(&SampleComponentManagerRequests::ResetRPIScene);
 
+            m_scene = AZ::RPI::RPISystemInterface::Get()->GetSceneByName(AZ::Name("RPI"));
+
```
Comment (invertednormal) [inline Gem/Code/Source/MSAA_RPI_ExampleComponent.cpp:105]: This should no longer be necessary. I fixed ResetScene() to update the scene directly a week or so ago. https://github.com/o3de/o3de-atom-sampleviewer/pull/488/

File: Gem/Code/Source/MultiSceneExampleComponent.cpp [hunk 2]
```diff
@@ -82,7 +82,7 @@ namespace AtomSampleViewer
         sceneDesc.m_featureProcessorNames.push_back("AZ::Render::PointLightFeatureProcessor");
         sceneDesc.m_featureProcessorNames.push_back("AZ::Render::PostProcessFeatureProcessor");
         sceneDesc.m_featureProcessorNames.push_back("AZ::Render::QuadLightFeatureProcessor");
-        sceneDesc.m_featureProcessorNames.push_back("ReflectionProbeFeatureProcessor");
```
Comment (invertednormal) [inline Gem/Code/Source/MultiSceneExampleComponent.cpp:85]: This has already been fixed. https://github.com/o3de/o3de-atom-sampleviewer/pull/488/

File: Gem/Code/Source/SampleComponentManager.cpp [hunk 3]
```diff
@@ -1546,7 +1546,7 @@ namespace AtomSampleViewer
         // Create and register the rhi scene with only feature processors required for AtomShimRenderer (only for AtomSampleViewerLauncher)
         RPI::SceneDescriptor sceneDesc;
         sceneDesc.m_nameId = AZ::Name("RHI");
-        sceneDesc.m_featureProcessorNames.push_back("AuxGeomFeatureProcessor");
```
Comment (invertednormal) [inline Gem/Code/Source/SampleComponentManager.cpp:1549]: This has already been fixed. https://github.com/o3de/o3de-atom-sampleviewer/pull/490

**Review by invertednormal — COMMENTED** [review]
I believe all of these fixes are already in in previous PRs. Not sure why this is even showing there as being diffs. See https://github.com/o3de/o3de-atom-sampleviewer/pull/488/ and https://github.com/o3de/o3de-atom-sampleviewer/pull/490

File: Gem/Code/Source/MSAA_RPI_ExampleComponent.cpp (hunk 1 above)
Comment (jiaweig-amzn) [reply Gem/Code/Source/MSAA_RPI_ExampleComponent.cpp:105]: I haven't updated the PR since last week. I'll close it.

**Pull request closed without merging**